src/qibo/gates/gates.py

```diff
@@ -2,11 +2,89 @@
 from typing import List
 
 import numpy as np
+import sympy as sp
 
 from qibo.config import PRECISION_TOL, raise_error
 from qibo.gates.abstract import Gate, ParametrizedGate
 
 
+class Parameter:
+    """Object which allows complex gate parameters. Several trainable parameter
+    and possibly features are linked through a lambda function which returns the
+    final gate parameter"""
```

Comment:
Here we should add the `Args:` section in the docstring and document the arguments of `__init__`.